kostal/__init__.py

```diff
@@ -18,88 +19,190 @@
 #  You should have received a copy of the GNU General Public License
 #  along with SmartHomeNG. If not, see <http://www.gnu.org/licenses/>.
 #
+#########################################################################
 
 import logging
+from lib.model.smartplugin import SmartPlugin
+from lib.utils import Utils
+import urllib.request, json
 import time
 import re
 
-logger = logging.getLogger('')
+class Kostal(SmartPlugin):
+    """
+    Since UI-version 6 the inverter can answere requests with json.
+    Unfortunately, I have only a simple inverter. Therefore, I can not test
+    the values for other phases or a second DC Line-In.
+    See README.md for more details
+    """
+    ALLOW_MULTIINSTANCE = True
 
+    PLUGIN_VERSION = "1.3.1.1"
 
-class Kostal():
+    _key2json = {
+       'operation_status' : 16780032,
+       'dctot_w' : 33556736,
+       'dc1_v' : 33555202,
+       'dc1_a' : 33555201,
+       'dc1_w' : 33555203,
+       'dc2_v' : 33555458,
+       'dc2_a' : 33555457,
+       'dc2_w' : 33555459,
+       'dc3_v' : 33555714,
+       'dc3_a' : 33555713,
+       'dc3_w' : 33555715,
+       'actot_w' : 67109120,
+       'actot_Hz' : 67110400,
+       'actot_cos' : 67110656,
+       'actot_limitation' : 67110144,
+       'ac1_v' : 67109378,
+       'ac1_a' : 67109377,
+       'ac1_w' : 67109379,
+       'ac2_v' : 67109634,
+       'ac2_a' : 67109633,
+       'ac2_w' : 67109635,
+       'ac3_v' : 67109890,
+       'ac3_a' : 67109889,
+       'ac3_w' : 67109891,
+       'yield_day_wh' : 251658754,
+       'yield_tot_kwh' : 251658753,
+       'operationtime_h' : 251658496
+    }
     _key2td = {
-        'power_current': 9,
-        'power_total': 12,
-        'power_day': 21,
-        'status': 27,
-        'string1_volt': 45,
-        'string2_volt': 69,
-        'string3_volt': 93,
-        'string1_ampere': 54,
-        'string2_ampere': 78,
-        'string3_ampere': 102,
-        'l1_volt': 48,
-        'l2_volt': 72,
-        'l3_volt': 96,
-        'l1_watt': 57,
-        'l2_watt': 81,
-        'l3_watt': 105
+        'actot_w': 9,
+        'yield_tot_kwh': 12,
+        'yield_day_wh': 21,
```

Comment:
The value for `yield_day_wh` in the HTML version is not `Wh`, it is `kWh` too:
```
Energie
Gesamtenergie 33386 kWh
Tagesenergie  26.58 kWh
```
I think we need a scale to specify the conversion (or unit-conversion methods) for generating always the same units for different devices.